### PR TITLE
Added biglake-catalog module

### PR DESCRIPTION
**Pull request by apichick**
Added biglake-catalog module

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/biglake-catalog/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Biglake Catalog
+
+This module allows to create a Biglake catalog with databases and corresponding tables in each database.
```
Comment (wiktorn) [inline modules/biglake-catalog/README.md:3]:
```suggestion
This module allows to create a BigLake Metastore with databases and corresponding tables in each database.
```
That's official naming according to the [docs](https://cloud.google.com/bigquery/docs/manage-open-source-metadata) - this also helps to understand what the module does (i.e. it doesn't provide ability to create external tables etc.)

File: modules/biglake-catalog/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            location_uri  = "gs://my-bucket/my-table-folder"
+            input_format  = "org.apache.hadoop.mapred.SequenceFileInputFormat"
+            output_format = "org.apache.hadoop.hive.ql.io.HiveSequenceFileOutputFormat"
+            parameters = {
+              "spark.sql.create.version"          = "3.1.3"
+              "spark.sql.sources.schema.numParts" = "1"
+              "transient_lastDdlTime"             = "1680894197"
+              "spark.sql.partitionProvider"       = "catalog"
+              "owner"                             = "John Doe"
+              "spark.sql.sources.schema.part.0"   = "{\"type\":\"struct\",\"fields\":[{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"name\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"age\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}"
```
Comment (wiktorn) [inline modules/biglake-catalog/README.md:45]:
```suggestion
    "spark.sql.sources.schema.part.0" = jsonencode({
      type = "struct"
      fields = [
        {
          name     = "id"
          type     = "integer"
          nullable = true
          metadata = {}
        },
        {
          name     = "name"
          type     = "string"
          nullable = true
          metadata = {}
        },
        {
          name     = "age"
          type     = "integer"
          nullable = true
          metadata = {}
        }
      ]
    })

```

**Review by wiktorn — COMMENTED** [review]
Left few comments. I think this is pretty niche resource, not sure whether this should go in, as the resource chain is pretty straightforward to build

File: modules/biglake-catalog/README.md (hunk 2 above)
Comment (apichick) [reply modules/biglake-catalog/README.md]: changed

File: modules/biglake-catalog/README.md (hunk 1 above)
Comment (apichick) [reply modules/biglake-catalog/README.md]: changed

**Comment by apichick** [comment]
changes done

**Review by wiktorn — APPROVED** [review]

**Comment by ludoo** [comment]
@apichick is this still planned for merge or should we close it?

**Pull request merged**